src/phpDocumentor/ConfigurationFactory.php

```diff
@@ -0,0 +1,104 @@
+<?php
+/**
+ * This file is part of phpDocumentor.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ *
+ * @copyright 2010-2015 Mike van Riel<mike@phpdoc.org>
+ * @license   http://www.opensource.org/licenses/mit-license.php MIT
+ * @link      http://phpdoc.org
+ */
+
+namespace phpDocumentor;
+
+use phpDocumentor\ConfigurationFactory\Strategy;
+
+/**
+ * The ConfigurationFactory converts the configuration xml from a Uri into an array.
+ */
+final class ConfigurationFactory
+{
+    /**
+     * The Uri that contains the path to the configuration file.
+     *
+     * @var Uri
+     */
+    private $uri;
+
+    /**
+     * The configuration xml.
+     *
+     * @var \SimpleXMLElement
+     */
+    private $xml;
+
+    /**
+     * All strategies that are used by the ConfigurationFactory.
+     *
+     * @var Strategy[]
+     */
+    private $strategies = [];
+
+    /**
+     * Initializes the ConfigurationFactory.
+     *
+     * @param Strategy[] $strategies
+     * @param Uri        $uri
+     * @param string     $schemaPath
+     */
+    public function __construct(array $strategies, Uri $uri, $schemaPath = '')
+    {
+        foreach ($strategies as $strategy) {
+            $this->registerStrategy($strategy);
+        }
+
+        $this->replaceLocation($uri);
+    }
+
+    /**
+     * Replaces the location of the configuration file if it differs from the existing one.
+     *
+     * @param Uri $uri
+     *
+     * @return void
+     */
+    public function replaceLocation(Uri $uri)
+    {
+        if ($this->uri !== $uri) {
+            $this->xml = new \SimpleXMLElement($uri, 0, true);
+
+            $this->uri = $uri;
+        }
+    }
+
+    /**
+     * Converts the phpDocumentor configuration xml to an array.
+     *
+     * @return array
+     *
+     * @throws \RuntimeException if no matching strategy can be found.
+     */
+    public function get()
+    {
+        foreach ($this->strategies as $strategy) {
```

Comment:
Protip (doesn't necessarily need to be changed at this stage): the `get()` method could be called multiple times during the life of the application but the parsing of the XML is a one-time thing. An optimalisation here would not be to cache the retrieved XML but to cache the converted array